src/main/kotlin/org/rust/lang/refactoring/RustNameSuggestions.kt

```diff
@@ -0,0 +1,101 @@
+package org.rust.lang.refactoring
+
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.codeStyle.NameUtil
+import com.intellij.psi.util.PsiTreeUtil
+import org.rust.ide.navigation.goto.RustSymbolNavigationContributor
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.util.childOfType
+import org.rust.lang.core.psi.util.parentOfType
+import org.rust.lang.core.types.util.resolvedType
+import java.util.*
+
+/**
+ * This suggests names for an expression about to be bound to a local variable
```

Comment:
@LiamClark what do yo think about this? Looks like a quite lot of places inside `RustIntroduceVariableRefactoring` would need to change from `PsiElement` to `RustExprElement` (like `findOccurrences` and `OccurrencesChooser.simpleChooser<RustExprElement>`), but this is actually a good idea to do anyway.